core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -1420,6 +1420,15 @@ class KafkaZkClient private[zk] (zooKeeperClient: ZooKeeperClient, isSecure: Boo
     zooKeeperClient.close()
   }
 
+  /**
+   * Wait up to the provided timeoutMs this KafkaZKClient to close.
+   * Supplying a timeoutMs of 0 means to wait forever.
+   */
+  def closeAndWait(timeoutMs: Int): Boolean = {
+    zooKeeperClient.closeAndWait(timeoutMs)
```

Comment:
Should we have this also remove the `ZooKeeperRequestLatencyMs` metric as the close() call?

Reply:
Thanks, I'll change this to use a private `close()` method shared by the public `close()` and `closeAndWait()` calls.